Formula/t/tth.rb

```diff
@@ -1,8 +1,14 @@
 class Tth < Formula
   desc "TeX/LaTeX to HTML converter"
-  homepage "http://hutchinson.belmont.ma.us/tth/"
-  url "http://hutchinson.belmont.ma.us/tth/tth_distribution/tth_4.16.tgz"
-  sha256 "ff8b88c6dbb938f01fe6a224c396fc302ae5d89b9b6d97f207f7ae0c4e7f0a32"
+  homepage "http://silas.psfc.mit.edu/tth/"
+  url "https://downloads.sourceforge.net/project/tth/tth4.16.tar.gz"
+  sha256 "b0e118d49a37e06598c1e2b524ea352ceabf064afef25acf02b556229ee43512"
+  license "GPL-2.0-only"
+
+  livecheck do
+    url "http://silas.psfc.mit.edu/tth/Version"
+    regex(/"v?(\d+(?:\.\d+)+)"/i)
```

Comment:
```suggestion
    url :stable
    regex(%r{url=.*?/tth[._-]?v?(\d+(?:\.\d+)+)\.t}i)
```

The formula's existing `stable` URL is from hutchinson.belmont.ma.us, so the previous `livecheck` block checking http://hutchinson.belmont.ma.us/tth/Version aligned with the `stable` source.

With the switch to a SourceForge tarball, we would normally check SourceForge for version information (livecheck has a `Sourceforge` strategy for this) rather than http://silas.psfc.mit.edu/tth/Version. The latter seems to be specifying the version of the unversioned files on http://silas.psfc.mit.edu/tth/tars/download.html and it may or may not correspond with the latest version on SourceForge.

Checking SourceForge for version information would make sure that livecheck only surfaces a new version when a related tarball is uploaded (which is typically how we approach this). The only potential downside is if upstream ever updates silas.psfc.mit.edu but forgets to update SourceForge, livecheck wouldn't pick up that version. That said, the opposite is also true, where there could be a new SourceForge version but livecheck wouldn't pick it up if silas.psfc.mit.edu wasn't updated yet.

Sometimes it's more appropriate to check the first-party site if upstream doesn't consider the version released until the site is updated (e.g., they may preemptively upload the files on SourceForge and then finalize the release by updating the first-party website). However, unless there's a special release process like this, we should simply check SourceForge here (aligning the check with the `stable` source).

Reply:
Many Thanks - I fully agree! It works - the livecheck does quite some magic, I didn't know it can check for the latest sourceforge version just having a download URL. I will squash the changes.